### PR TITLE
Rare Spider Variants!

### PR DESCRIPTION
**Pull request by ghost**
:cl:
add: Rare Spider Variants have arrived! Every spiderling has a rare chance of growing up to a rare version of their original spider type!
add: Special thanks to Onule for the sprites, you're the best!
/:cl:

# SPIDERS HAVE BEEN GIVEN THE LOVE THEY DESERVE!

## New spider sprites!
NURSE =
nurse
HUNTER =
hunter
GUARD = 
guard
### Very rarely, a spiderling will become an evolved version of one of these creatures!
MIDWIFE = Gains a command function to message all spiders, can remotely kill off spiders. Praise the queen!
nursenew
VIPER = Lower health than the hunter and almost no melee damage but an **extremely** poisonous bite!
hunternew
TARANTULA = Massive hulking spider that does crazy damage but moves slow off of webs. Great for protecting territory, will get shotgunned down outside of it.
guardnew

### CODE REVIEW

**Comment by Armhulen** [comment]
ALSO PLANNED: Giving text explaining what each spider does so you don't get confused when you get a special variant

**Comment by Ratvar** [comment]
Make vipers slowly generate poison for their attack, building over time? Make them inject acid into bloodstream? Fast 2hit(1hit, to be honest)death from speedy spider is not fun.
Queen-ability to announce things?
Max spider web per tile, so goliaths cannot spam one tile forever? Make webs collide into webby walls, so they can prevent fires and such, when 10 webs are on the same tile?

**Comment by RemieRichards** [comment]
Nurse > Midwife
Giant > Tarantula

**Comment by Armhulen** [comment]
i like it remie, i'll switch it

**Comment by XDTM** [comment]
are spiders the new golems? count me in

**Comment by Armhulen** [comment]
@PraiseRatvar the spider injects a toxin that will kill you EVENTUALLY if you don't go to medical, but it doesn't kill you any faster than a hunter. A hunter will actually kill you way faster with their 15 force attacks and 120 health.

@XDTM don't you dare >:(

**Comment by XDTM** [comment]
Also, 80 poison is like ~7 minutes of 2 poison damage per tick, for a total of 400 poison damage per bite, if it's actually 80 units of reagent

**Comment by Anonmare** [comment]
You could look up Paradise's Terror Spiders for inspiration
https://nanotrasen.se/wiki/index.php/Terror_Spider

**Comment by DoctoreMad** [comment]
Daily reminder that just as 10 seconds is longer than you think and 7 minutes is shorter than you think

**Comment by Armhulen** [comment]
@XDTM bring charcoal!

**Comment by XDTM** [comment]
@DoctoreMad it's more than 10% of an average round

per bite

**Comment by DoctoreMad** [comment]
Er, what does that mean?

**Comment by XDTM** [comment]
counting an average round as 60 minutes, it's 7 minutes of poison per bite, aka a bit more of 10% of it

**Comment by Armhulen** [comment]
you're basically immune to the effects if you have charcoal in your system though, i promise!

**Comment by XDTM** [comment]
that's a best case scenario though, most won't have any

**Comment by XDTM** [comment]
IMO it should be a normal amount of a stronger poison

**Comment by Armhulen** [comment]
fighting spiders without charcoal is like fighting shadowlings without light? anyways i'll tone it down to 60

**Comment by Armhulen** [comment]
i'd kinda like that but then i have to redo some of the var spider code and aaaaaaaaaaaaaaaaaaaaaauuuuuuuuuuuuuuuuuuuuuuuuuuuggggggggggggggggggghhhhhhhhhhh

**Comment by XDTM** [comment]
not really, there's a var for the reagent to use as toxin, it's quite simple

**Comment by Armhulen** [comment]
wait let me look at it again

**Comment by Armhulen** [comment]
hoy fuck that's easy to change, any ideas for what toxin to use

**Comment by XDTM** [comment]
there is the appropriately-named venom, lemme see how strong it is

**Comment by XDTM** [comment]
power increases with the amount inside the target, seems ideal! Repeated bites means faster death, instead of a longer recovery period

**Comment by Armhulen** [comment]
xtdm we're going places, i'll make that the venom

**Comment by XDTM** [comment]
10-15 per bite should be balanced, if the bite itself doesn't do damage

**Comment by XDTM** [comment]
by the way, move_to_delay only applies to ai movement, i'm pretty sure, which means that player tarantulas would wreck shit

**Comment by Armhulen** [comment]
i think you're wrong? all the spiders had it before and they all move at different speeds.

**Comment by XDTM** [comment]
`var/move_to_delay = 3 //delay for the automated movement.`

**Comment by DoctoreMad** [comment]
Isn't move_to_delay the delay for idle(non-combat) movement?
(Note: noncoder)

**Comment by Armhulen** [comment]
i see, let me test player speeds right now. also, how would you make it so you can't push a mob like juggernauts and goliaths

**Comment by XDTM** [comment]
it's for AI-controlled movement

**Comment by Armhulen** [comment]
>how would you make it so you can't push a mob like juggernauts and goliaths

**Comment by XDTM** [comment]
lemme see;

meanwhile, set `speed = 3` (or higher/lower) to slow down the taratulas controlled by players

**Comment by Armhulen** [comment]
okay thanks

**Comment by XDTM** [comment]
to prevent pushing, set `status_flags = NONE`;
also add `mob_size = MOB_SIZE_LARGE`

**Comment by Armhulen** [comment]
guess i'll start on the midwife, really good at making bases and can send messages to all spiders.

**Comment by Armhulen** [comment]
also tell me if the tarantula values are too crazy.

**Comment by ExcessiveUseOfCobblestone** [comment]
sprites tho?

**Comment by Armhulen** [comment]
i need sprites, i'll do them myself but i'm hoping someone comes along and does an amazing job

**Comment by ExcessiveUseOfCobblestone** [comment]
the trick is to do your own then a spriter will get mad and make a good sprite for you.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 1]
```diff
@@ -103,6 +103,47 @@
...
+	icon_living = "hunter"
+	icon_dead = "hunter_dead"
+	maxHealth = 40
+	health = 40
+	melee_damage_lower = 1 //what doesn't kill you...
+	melee_damage_upper = 1
+	poison_per_bite = 15
+	move_to_delay = 4
+	poison_type = "venom" //will soon after if there isn't medical attention!
+	speed = 4
```
Comment (XDTM) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: Higher speed means slower movement, since it's actually the delay between steps

**Comment by Armhulen** [comment]
aaaaah more things breaking one seconddddd

**Comment by Jordie0608** [comment]
But tarantulas are a small species

**Comment by Armhulen** [comment]
i think it's the idea not the


i mean we don't have real spiders and the handwives aren't literally hand wives right

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 2]
```diff
@@ -295,6 +336,36 @@
...
+	if(!input || !IsAvailable())
+		return
+
+	spider_command(usr, input)
+
+/proc/spider_command(mob/living/user, message)
+	var/my_message
+	if(!message)
+		return
+	my_message = "<b>COMMAND FROM SPIDER QUEEN:</b> [message]
```
Comment (kevinz000) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: missing "

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: i wouldn't review this, i just fixed a huge part of it don't worry

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: i, as in cobby and oranges

**Comment by Armhulen** [comment]
okay lets test this
i'm excited!!

**Comment by Armhulen** [comment]
okay the code is working and it is working like a charm

now lets get to the cool part, new sprites

**Comment by SkowronX** [comment]
>The spider can wiggle dance.

**Comment by Armhulen** [comment]
> The spider can wiggle dance.

strut my stuff my stuff is so SHINY!

**Comment by ShadowDeath6** [comment]
I would not recommend venom since every time it metabolizes it has a 15% chance to decay into histamine(between 5-10 for 1.1 venom) which has an absurdly lethal overdose process regardless of time spent in the system

https://github.com/tgstation/tgstation/blob/master/code/modules/reagents/chemistry/reagents/toxin_reagents.dm#L406

**Comment by Armhulen** [comment]
okay, any better ideas

**Comment by Armhulen** [comment]
okay that's it, the new viper toxin is spewium

**Comment by ShadowDeath6** [comment]
possibly a weaker version of amanitin if you still want to go with the "kills you the longer it's in you" route

**Comment by Armhulen** [comment]
>making a whole new toxin for a spider that has a 1/100 chance of spawning every time a spiderling grows up

**Comment by ShadowDeath6** [comment]
go ahead and use venom then, if it's that rare I doubt it'll be much of an issue anyway

**Comment by Armhulen** [comment]
i'm probably going to use spewium for it's meme value

**Comment by Armhulen** [comment]
nope nevermind stun combat sux

**Comment by FantasticFwoosh** [comment]
With a bit more development, these could be a suitable replacement for xenomorphs like how paradise has terror spiders to fill in that role.

**Comment by RemieRichards** [comment]
We dont need to replace anything you stupid fwoosh.

**Comment by Bawhoppen** [comment]
I hate our spider sprite theme. Bay's spiders look much more menacing and cool.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 3]
```diff
@@ -103,6 +103,53 @@
...
+/mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife
+	desc = "Furry and black, it makes you shudder to look at it. This one has scintillating green eyes."
+	icon_state = "midwife"
+	icon_living = "midwife"
+	icon_dead = "midwife_dead"
+	maxHealth = 40
+	health = 40
+	var/datum/action/innate/spider/comm/letmetalkpls
+
+	/mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife/Initialize()
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: indentation

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 4]
```diff
@@ -103,6 +103,53 @@
...
+		..()
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: . = ..()

File: code/_globalvars/lists/mobs.dm [hunk 5]
```diff
@@ -20,6 +20,7 @@ GLOBAL_LIST_EMPTY(ai_list)
 GLOBAL_LIST_EMPTY(pai_list)
 GLOBAL_LIST_EMPTY(available_ai_shells)
 GLOBAL_LIST_EMPTY(simple_animals)
+GLOBAL_LIST_EMPTY(spidermobs)				//all spider mobs
```
Comment (Cyberboss) [inline code/_globalvars/lists/mobs.dm]: these are only logged in spider mobs, fix the comment or change the functionality

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 6]
```diff
@@ -295,6 +342,39 @@
...
+
+/datum/action/innate/spider/comm/Trigger()
+	var/input = stripped_input(usr, "Input a message for your legions to follow.", "Command", "")
+	if(QDELETED(src) || !input || !IsAvailable())
+		return FALSE
+	spider_command(usr, input)
+	return TRUE
+
+/datum/action/innate/spider/comm/proc/spider_command(mob/living/user, message)
+	var/my_message
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: why is this declared up here?

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 7]
```diff
@@ -295,6 +342,39 @@
...
+	if(!message)
+		return
+	my_message = "<b>COMMAND FROM SPIDER QUEEN:</b> [message]"
+	for(var/mob/living/simple_animal/hostile/poison/giant_spider/M in GLOB.spidermobs)
+		to_chat(M, my_message)
+	log_talk(user, "SPIDERCOMMAND:[key_name(user)] : [message]",LOGSAY)
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: add a space after the first colon

File: code/_globalvars/lists/mobs.dm (hunk 5 above)
Comment (Cyberboss) [reply code/_globalvars/lists/mobs.dm]: preferably the latter

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/_globalvars/lists/mobs.dm (hunk 5 above)
Comment (ExcessiveUseOfCobblestone) [reply code/_globalvars/lists/mobs.dm]: Its only use is to message players [as of now at least] , it should only be Player-controlled spiders.

Comment (Cyberboss) [reply code/_globalvars/lists/mobs.dm]: Yes, but it's much saner to keep track of all spiders and who cares if AI mobs get the message?

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 6 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: he had some thing for deadchat last night where it was needed but I removed it without removing the var, whoops :p

File: code/_globalvars/lists/mobs.dm (hunk 5 above)
Comment (ExcessiveUseOfCobblestone) [reply code/_globalvars/lists/mobs.dm]: fair enough

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 6 above)
Comment (Cyberboss) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: Yeah, so move it down to there

Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: you are too smart for your own good cyberman!

**Comment by Armhulen** [comment]
I'll fix all that when I have time, I'm probably going to go with 30u of polonium as it's slow acting but really deadly

**Comment by DoctoreMad** [comment]
Instead of inserting just one toxin don't you think that you can add one base reagent that deals low toxin damage and another one that starts dealing damage after a amount of time passes?
It would also halve the time the poison stays in the target since charcoal will proc for both chemicals. At least that's how I think it works.

**Comment by Armhulen** [comment]
Currently the code only allows for one kind of poison on the bite. I could MAKE a chemical that does that but it seems like a slog to do.

File: code/_globalvars/lists/mobs.dm (hunk 5 above)
Comment (Armhulen) [reply code/_globalvars/lists/mobs.dm]: i'll get around to this zzz

**Comment by Armhulen** [comment]
Now hol up I'm getting sprites in and I need to add a spawn chance of these

**Comment by XDTM** [comment]
Chemicals are pretty easy, also venom is still the right choice imo

**Comment by FantasticFwoosh** [comment]
@RemieRichards watch how fast WJohn vanishes when xeno's are washed out of the code in favour of softer spider hive swarms.

**Comment by Armhulen** [comment]
@XDTM I'll make a custom poison that is venom but without the decay into deathjuice

**Comment by ShadowDeath6** [comment]
https://github.com/tgstation/tgstation/pull/30593#issuecomment-328398410

:^)

**Comment by Armhulen** [comment]
I'm not happy about it, I'm only doing it because it'll compile!

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 8]
```diff
@@ -4,7 +4,7 @@
 #define MOVING_TO_TARGET 3
 #define SPINNING_COCOON 4
 
-/mob/living/simple_animal/hostile/poison
+/mob/living/simple_animal/hostile/poison //this guy's a fukkin idiot everyone knows that spiders inject venom not poison
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: It's a subtype of hostile mobs that transfer reagents and aren't exclusive to spiders.

See: Bees.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 9]
```diff
@@ -103,6 +103,53 @@
 	poison_per_bite = 5
 	move_to_delay = 5
 
+//vipers are the rare variant of the hunter, no damage but so much poison medical care will be needed fast. Low hp, high speed and no damage so hit and run tactics.
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: They DO have damage though!

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 10]
```diff
@@ -103,6 +103,53 @@
...
+	icon_state = "hunter"
+	icon_living = "hunter"
+	icon_dead = "hunter_dead"
+	maxHealth = 40
+	health = 40
+	melee_damage_lower = 1
+	melee_damage_upper = 1
+	poison_per_bite = 30
+	move_to_delay = 4
+	poison_type = "polonium" //slow acting but this at 30u per bite will seriously mess you up, perfect viper poison
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: This gives RADS

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 11]
```diff
@@ -103,6 +103,53 @@
...
+	speed = 4
+
+//tarantulas are really tanky, regenerating (maybe), hulky monster but are also extremely slow, so.
+/mob/living/simple_animal/hostile/poison/giant_spider/tarantula
+	desc = "Furry and black, it makes you shudder to look at it. This one has abyssal red eyes."
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: no name difference?

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 12]
```diff
@@ -103,6 +103,53 @@
...
+	icon_state = "guard"
+	icon_living = "guard"
+	icon_dead = "guard_dead"
+	maxHealth = 400 // woah nelly
+	health = 400
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: Just a heads up this is almost twice the health of a Juggernaut.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 13]
```diff
@@ -103,6 +103,53 @@
...
+	melee_damage_lower = 35
+	melee_damage_upper = 40
+	poison_per_bite = 0
+	move_to_delay = 8
+	speed = 3
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: This is not going to be fun to play as.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 14]
```diff
@@ -103,6 +103,53 @@
...
+	status_flags = NONE
+	mob_size = MOB_SIZE_LARGE
+
+//midwives are the queen of the spiders, can send messages to all them and web faster. That rare round where you get a queen spider and turn your 'for honor' players into 'r6siege' players will be a fun one.
+/mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: no name difference?

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 15]
```diff
@@ -295,6 +342,39 @@
 		busy = SPIDER_IDLE
 		stop_automated_movement = FALSE
 
+/mob/living/simple_animal/hostile/poison/giant_spider/Login()
+	..()
+	GLOB.spidermobs += src
+
+/mob/living/simple_animal/hostile/poison/giant_spider/Logout(GLOB.spidermobs)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: Remove this Arg

**Review by ExcessiveUseOfCobblestone — CHANGES_REQUESTED** [review]
That Tarantula is not going to be fun to play as or against.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 8 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: hahahaha bees still inject venom haHA

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 9 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: one damage cmon now

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 10 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: yeah, rads are slow acting and do horrible effects if you don't get medical attention, not a terrible one for the viper

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 11 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: deep > abyssal

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 12 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: 200 health on the normal spider is already giant, i'll edit this down to 300 though good point

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 13 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: i could make them fast on webs. oh dear me!

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 15 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: caaaaan do

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 14 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: no? midwife?

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: i don't know what you mean here

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 11 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: EYE COLOR is what i mean

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 10 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: i'm making a new one though

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 14 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: you didn't set a name var.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 11 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: just like midwife you may want to add `name = "tarantula"`

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 9 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: and rads/toxin damage...

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 14 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: why are we giving them special names? Oh whatever, it'll be kinda nice when it happens.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 11 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: fine fine

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 9 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: okay fine i'll reword it

**Comment by Armhulen** [comment]
whoops i should be fucking compiling lmao

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm [hunk 16]
```diff
@@ -445,6 +445,66 @@
...
+	name = "Viper Venom"
+	id = "venom2"
+	description = "Venom extracted from a viper. Does low damage and takes a long time to go away, but left untreated will slowly advance and get worse." //Charcoal, dammit!
+	reagent_state = LIQUID
+	color = "#F0FFF0"
+	metabolization_rate = 0.1 * REAGENTS_METABOLISM
+	toxpwr = 0
+
+/datum/reagent/toxin/spidervenom/on_mob_life(mob/living/M) //this is a big deal, the venom works like a virus kinda
+	var/stage = 1
```
Comment (Cyberboss) [inline code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: You wrote this like a disease >.>

**Review by Cyberboss — DISMISSED** [review]

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 16 above)
Comment (Armhulen) [reply code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: is that bad?

Comment (Armhulen) [reply code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: if it works ¯\_(ツ)_/¯

**Comment by Armhulen** [comment]
ultra wip on the venom, haha...

**Comment by Armhulen** [comment]
also i just discovered switch so this code is about to get a lot cooler

**Comment by DoctoreMad** [comment]
> can remotely kill off spiders

HA HA TIME TO GRIEF is this logged

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 14 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: You're the one giving them hierarchical structure, would make sense for the top dog to be a different name than the common peon.

File: code/modules/reagents/chemistry/reagents/toxin_reagents.dm (hunk 16 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/toxin_reagents.dm]: needs to be a switch

**Comment by Armhulen** [comment]
One in 100, you could grief way easier with other spiders

**Comment by Armhulen** [comment]
okay not with other spiders but with like, a maxcap. every round basically for that

**Comment by Ratvar** [comment]
admin notify with rejenivate button every time spider queen kills someone?

**Comment by WJohn** [comment]
These sprites kick ass!

**Comment by ExcessiveUseOfCobblestone** [comment]
A changelog thanking onule would be really sweet.

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 14 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: I wish I knew how to remove this entire section since it was changed :(

**Comment by Armhulen** [comment]
sounds good cobblestone

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm (hunk 14 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: i could change the pr then change it back that would close this right

**Comment by Armhulen** [comment]
I'll cut the remote execution for another pr, this is almost done!

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 17]
```diff
@@ -103,6 +103,63 @@
...
+	health = 300
+	melee_damage_lower = 35
+	melee_damage_upper = 40
+	poison_per_bite = 0
+	move_to_delay = 8
+	speed = 7 //changing this to be fast on webs, so don't hate!
+	status_flags = NONE
+	mob_size = MOB_SIZE_LARGE
+
+/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/Life(mob/living/M)
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]:
```
/mob/living/simple_animal/hostile/poison/giant_spider/tarantula/movement_delay()
	var/turf/T = get_turf(src)
	if(locate(/obj/structure/spider/stickyweb) in T)
		speed = 2
	else
		speed = 7
	. = ..()
```

File: code/game/objects/effects/spiders.dm [hunk 18]
```diff
@@ -171,15 +180,18 @@
...
-			S.faction = faction.Copy()
-			if(player_spiders)
-				S.playable_spider = TRUE
-				notify_ghosts("Spider [S.name] can be controlled", null, enter_link="<a href=?src=\ref[S];activate=1>(Click to play)</a>", source=S, action=NOTIFY_ATTACK)
-			qdel(src)
+				if(prob(3))
+					grow_as = pick(/mob/living/simple_animal/hostile/poison/giant_spider/tarantula, /mob/living/simple_animal/hostile/poison/giant_spider/hunter/viper, /mob/living/simple_animal/hostile/poison/giant_spider/nurse/midwife)
+				else
+					grow_as = pick(/mob/living/simple_animal/hostile/poison/giant_spider, /mob/living/simple_animal/hostile/poison/giant_spider/hunter, /mob/living/simple_animal/hostile/poison/giant_spider/nurse)
+				var/mob/living/simple_animal/hostile/poison/giant_spider/S = new grow_as(src.loc)
```
Comment (ChangelingRain) [inline code/game/objects/effects/spiders.dm]: you've got some fucked indentation here; this and everything below it should all be tabbed down one

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 19]
```diff
@@ -295,6 +353,35 @@
 		busy = SPIDER_IDLE
 		stop_automated_movement = FALSE
 
+/mob/living/simple_animal/hostile/poison/giant_spider/Login()
+	..()
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: . = ..()

File: code/modules/mob/living/simple_animal/hostile/giant_spider.dm [hunk 20]
```diff
@@ -295,6 +353,35 @@
 		busy = SPIDER_IDLE
 		stop_automated_movement = FALSE
 
+/mob/living/simple_animal/hostile/poison/giant_spider/Login()
+	..()
+	GLOB.spidermobs += src
```
Comment (ChangelingRain) [inline code/modules/mob/living/simple_animal/hostile/giant_spider.dm]: add this somewhere and make this line `GLOB.spidermobs[src] = TRUE`
```
/mob/living/simple_animal/hostile/poison/giant_spider/Destroy()
	GLOB.spidermobs -= src
	return ..()
```

**Pull request merged**